Source/Diagnostics/ParticleIO.cpp

```diff
@@ -127,11 +129,26 @@ MultiParticleContainer::WritePlotFile (const std::string& dir) const
 
             // Convert momentum to SI
             pc->ConvertUnits(ConvertDirection::WarpX_to_SI);
+
+            // build filter functors
+            bool do_random_filter = true;
+            bool do_uniform_filter = false;
+            Real random_fraction = 0.25;
+            long uniform_stride = 3;
+
+            RandomFilter const random_filter(do_random_filter, random_fraction);
+            UniformFilter const uniform_filter(do_uniform_filter, uniform_stride);
+
             // real_names contains a list of all particle attributes.
             // pc->plot_flags is 1 or 0, whether quantity is dumped or not.
-            pc->WritePlotFile(dir, species_names[i],
-                              pc->plot_flags, int_flags,
-                              real_names, int_names);
+            pc->WritePlotFile(
+                dir, species_names[i],
+                pc->plot_flags, int_flags,
+                real_names, int_names,
+                [=] (const ParticleType& p)
+                {
+                    return random_filter(p) * uniform_filter(p) ;
```

Comment:
selecting this in-kernel is a bit expensive in registers (always includes rng logic) but probably not to worry about if the overall kernel is small.
Otherwise we can add the `m_is_active` check before calling `WritePlotFile` and add this twice (probably having both active at the same time is not necessary).

Reply:
Let's keep that In mind, and update if it starts to have a big footprint.